### PR TITLE
Tweaks Deathsquad

### PR DESCRIPTION
**Pull request by Kyep**
🆑 Kyep
tweak: DeathSquad is now better equipped, and prompts eligible ghosts with a "Do you want to play as DeathSquad?" when called.
add: DeathSquad members (except team leaders) can now choose to spawn as either an organic DS member, or a DS borg.
/🆑

Changes you may notice:
- Upon pressing the DS/SST button, the admin gets an alert. This alert now correctly says that the FIRST person spawned will be the team leader. 
- After dismissing this notice, an '_(admin name) has started to spawn a (team type)_' notice is broadcast to all online admins. This is necessary to stop multiple admins sending a team at once (teams can take time to spawn, due to changes detailed later).
- At this point, ghosts are evaluated to see who is eligible to be DS/SST. A minimum of 10 hours of playtime (same as ERT) is required to be eligible. Unlike previously, admins are now eligible (though, they can hit NO on the prompt like anyone else).
- All eligible ghosts get a popup: "_Do you want to join the DeathSquad/Syndicate Strike Team?_". You have to hit 'YES' on this to be considered. Obviously, this check excludes anyone who is AFK, or unwilling.
- After we know who is willing on the player side, the summoning admin is presented with a list of willing players (by ckey). The admin chooses which players get selected. It is NOT random. Though, the admin can simply hit 'OK' a bunch of times to choose whoever is first (said yes first) in the list.
- At this point, calling the DS previously auto-recalled the shuttle. This no longer happens.
- Now, actual DS/SST start to spawn. In both cases, the FIRST person selected really will be designated team leader. Previously, the LAST person spawned was team leader, but this was bad, because there is no guarantee that person had a player behind the mob, which led to diskless DS.
- All organic members of DS now get pinpointers. 
- Non-leader members of DS are now offered the chance to play as a DS borg. DS borgs have pulse guns, eswords, crowbars, emag/flash/flashbang immunity, and will wreck you. They get copies of the nuke code and mission text. They cannot activate the nuke themselves, which is why the DS team leader is not eligible to be one (he has to be able to arm the nuke). Since each DS member (except leader) must choose organic or synthetic DS during DS spawn, the spawning process is now wrapped in a _spawn(0)_ so that a single afk DS member will not delay half the team spawning.
- Organic DS members now get black jumpsuits, not green.
- All DS and SST members now get a reminder of their role and (admin-set) mission in big, colorful text.

Changes you probably won't notice:
- Lots of back-end refactoring. Replaced 'leader_selected' (number) with is_leader (boolean). Rewrote "if (!X)" conditions to put the "if (X)" case first. Moved the grabbing of ckeys into a new proc that is shared between DS/SST, fixed a bunch of 'font color's, and more.

ds_modules
ds_intro_text

### CODE REVIEW

**Comment by granodd** [comment]
Deathsquad borgs should also be given what engineering borgs have, which is that mag pulse code/system that makes them immune to fastmos. Otherwise the holes they're making into space from shooting walls would just screw them over, as I've seen happen on a few occasions (though i might be mistaking that for the time Nation Droidekas were spawned for the deathsquad against terror spiders)

**Comment by Allfd** [comment]
Syndi and DS borgs work the same as engineering borgs.

**Comment by Spartan6** [comment]
Perhaps make the borgs, when flashed, not get stunned but have their modules reset and they need to select them again? Don't want such an easy counter to the deadliest tech NT has.

[borg name] seems to recoil from the flash, their modules resetting!

File: code/modules/admin/verbs/striketeam.dm [hunk 1]
```diff
@@ -40,42 +40,71 @@ var/global/sent_strike_team = 0
...
+					if(!is_leader)
+						var/new_gender = alert(src, "Select Deathsquad Type.", "DS Character Generation", "Organic", "Cyborg")
+						if(new_gender == "Cyborg")
+							use_ds_borg = TRUE
+
+				if(use_ds_borg)
+					var/mob/living/silicon/robot/deathsquad/R = new()
+					R.forceMove(get_turf(L))
+					var/rnum = rand(1,1000)
+					var/borgname = "Specops [rnum]"
```
[the remark this answers is not in the document]
Comment (Kyep) [reply code/modules/admin/verbs/striketeam.dm]: Changed.

**Comment by Kyep** [comment]
@Spartan6 How exactly are you going to get close enough to a DS borg to flash it? Won't it just shoot you from range with its built-in pulse rifle? Even if you do flash it, won't its friends just kill you? I could add code to prevent flashes working on DS borgs, but it seems like a really rare case, and I'm not sure its worth it.

**Comment by Spartan6** [comment]
You can argue that with it's ERT brethren. Won't they just shoot you from range? It happens. They might split up to cover the most ground.

Normal Death Squad are flash proof, the borgs should be too.

**Comment by davidchan13** [comment]
>@Kyep How exactly are you going to get close enough to a DS borg to flash it?

Flashbangs.

**Comment by Kyep** [comment]
@Spartan6 @davidchan13 DS borgs, like organic DS officers, are now considered to have ear/eye protection. This makes them immune to flashes and flashbangs.

**Comment by KingPhilipIII** [comment]
Can you bring Deathsquads up to speed and ask players if they want to be squaddies instead of grabbing from players without warning?

**Comment by Kyep** [comment]
KingPhilipIII: Done. PR description updated. Both SST and DS now prompt players to check who is willing, and the admin picks from willing players - rather than just yanking ghosts without warning.

**Comment by KingPhilipIII** [comment]
:okhand:

**Comment by Spartan6** [comment]
@Kyep Law 2 and Law 3 distinguishes that CC officials and death commandos are not the same thing, but Law 1 only forbids harming the CC officials. This means the borgs would be able to freely eliminate Death Commandos.

Law 5 should also be changed from the wording ‘crewmember’. Only Death Squad should survive, nobody else unless ordered, crewmember or not.

**Comment by Kyep** [comment]
DS Borg lawset is now:
1) You may not injure a Central Command official or, through inaction, allow a Central Command official to come to harm.
2) You must obey orders given to you by Central Command officials.
3) You must work with your commando team to accomplish your mission.

In practice, this means that DS borgs should default to carrying out their mission at all times, unless given explicit orders otherwise by an admin.

**Comment by Kyep** [comment]
Closing & Reopening to force Travis recheck.

**Comment by Allfd** [comment]
One squad was left out....  I will take care of it after this merges.

**Pull request merged**